sdk/nodejs/resource.ts

```diff
@@ -771,8 +772,14 @@ export class ComponentResource extends Resource {
      * @param unused [Deprecated].  Component resources do not communicate or store their properties
      *               with the Pulumi engine.
      * @param opts A bag of options that control this resource's behavior.
-     */
-    constructor(type: string, name: string, unused?: Inputs, opts: ComponentResourceOptions = {}) {
+     * @param asyncConstructed If this is a component that will construct its children
+     * asynchronously. Components that do this *must* call [registerOutputs] to signal when
+     * construction is complete.  If [registerOutputs] is not called then hangs will occur if
+     * another resource 'dependsOn' this resource.
+     */
+    constructor(type: string, name: string, unused?: Inputs,
+                opts: ComponentResourceOptions = {},
+                asyncConstructed: boolean = false) {
```

Comment:
note: i have implemented this as a constructor param.  That's because it felt "wrong" somehow to be an "opt".  It's not something the caller thinks about when constructing an Component,